pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -0,0 +1,652 @@
+		args         []string
+	)
+	for _, secret := range secrets {
+		switch {
+		case !filter(secret):
+			continue
+		case secret.Type == corev1.SecretTypeBasicAuth && secret.Annotations[DOCKERSecretAnnotationPrefix] != "":
+			args = append(args,
+				fmt.Sprintf("-basic-%s=%s=%s", "docker", secret.Name, secret.Annotations[DOCKERSecretAnnotationPrefix]))
+		case secret.Type == corev1.SecretTypeDockerConfigJson:
+			args = append(args, fmt.Sprintf("-dockerconfig=%s", secret.Name))
+		case secret.Type == corev1.SecretTypeDockercfg:
+			args = append(args, fmt.Sprintf("-dockercfg=%s", secret.Name))
+		case secret.Type == corev1.SecretTypeBasicAuth && secret.Annotations[GITSecretAnnotationPrefix] != "":
+			annotatedUrl := secret.Annotations[GITSecretAnnotationPrefix]
+			args = append(args, fmt.Sprintf("-basic-%s=%s=%s", "git", secret.Name, annotatedUrl))
+		case secret.Type == corev1.SecretTypeSSHAuth:
+			annotatedUrl := secret.Annotations[GITSecretAnnotationPrefix]
+			args = append(args, fmt.Sprintf("-ssh-%s=%s=%s", "git", secret.Name, annotatedUrl))
+		default:
+			//ignoring secret
+			continue
+		}
+
+		volumeName := fmt.Sprintf(SecretTemplateName, secret.Name)
+
+		volumes = append(volumes, corev1.Volume{
+			Name: volumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: secret.Name,
+				},
+			},
+		})
+
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      volumeName,
+			MountPath: fmt.Sprintf(SecretPathName, secret.Name),
+		})
+	}
+
+	return volumes, volumeMounts, args
+}
+
+func (b *Build) setupBindings(bindings []ServiceBinding) ([]corev1.Volume, []corev1.VolumeMount) {
+	volumes := []corev1.Volume{}
+	volumeMounts := []corev1.VolumeMount{}
+
+	for _, s := range bindings {
+		if s.SecretRef != nil {
+			secretVolume := fmt.Sprintf("service-binding-secret-%s", s.Name)
+			volumes = append(volumes,
+				corev1.Volume{
+					Name: secretVolume,
+					VolumeSource: corev1.VolumeSource{
+						Secret: &corev1.SecretVolumeSource{
+							SecretName: s.SecretRef.Name,
+						},
+					},
+				},
+			)
+			volumeMounts = append(volumeMounts,
+				corev1.VolumeMount{
+					Name:      secretVolume,
+					MountPath: filepath.Join(platformVolume.MountPath, "bindings", s.Name),
+					ReadOnly:  true,
+				},
+			)
+		}
+		// If this is a v1alpha1 binding
+		if s.V1Alpha1MetadataRef != nil {
+			metadataVolume := fmt.Sprintf("service-binding-metadata-%s", s.Name)
+			volumes = append(volumes,
+				corev1.Volume{
+					Name: metadataVolume,
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: *s.V1Alpha1MetadataRef,
+						},
+					},
+				},
+			)
+			volumeMounts = append(volumeMounts,
+				corev1.VolumeMount{
+					Name:      metadataVolume,
+					MountPath: filepath.Join(platformVolume.MountPath, "bindings", s.Name, "metadata"),
+					ReadOnly:  true,
+				},
+			)
+		}
+	}
```

Comment:
There is no "metadata" directory in the k8s service binding spec, and the secret is in the wrong place for the cnb service binding spec.

rm unless this is doing something meaningful that I don't understand.

Reply:
yea I did this wrong, fixing

Reply:
this should be fixed now